documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+
+There are many kinds of breaking changes that can ship in the .NET SDK, such as:
+
+* New MSBuild warnings and errors (props/targets)
+* New NuGet warnings and errors. 
+  * For example, NuGet Audit.
+* Roslyn Analyzers and CodeFixes
+  * This includes trimming/ILLink analyzers and codefixes
+* Behavioral/implementation changes
+  * MSBuild engine changes like MSBuild Server
+  * Implementation changes for MSBuild Tasks
+  * NuGet Restore algorithm enhancements
+  * Changes to DotNet CLI grammar
+  * Changes to defaults in CLI flags that impact behavior
+    * For example, `--configuration` flag defaulting to `Release` instead of `Debug`
+
+## Configuration Knobs
+
+The following knobs are available to enable/disable these changes (some may not apply to all kinds of changes):
+
+* [TargetFramework](https://learn.microsoft.com/en-us/dotnet/standard/frameworks)
+* [SdkAnalysisLevel](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#sdkanalysislevel)
+* EditorConfig (for Analyzers)
+* AnalysisLevel (for Analyzers/CodeFixes)
+* WarningLevel
+
+## Deciding how to Surface Changes
+
+### Implement changes in an informational/non-blocking way initially
+
+What this means will vary change-to-change. For example, for a change expressed as an Analyzer or MSBuild diagnostic, consider
+Informational level severities initially. For a behavioral change on a CLI, consider an informational message written to the
+stderr channel on the console instead of making the stdout output un-parseable by tools.
+
+Concrete example: When the `dotnet new --list` command was changed to `dotnet new list` to adhere to CLI design guidelines,
+the old command was still the default supported and a warning was written to the console when the old form was used pointing users to the new form. This allowed users and scripts to continue using the old form and gradually migrate to the new.
+
+### Gradually increase severity over each release
+
+If a change in introduced in an informational/non-blocking way, determine the time frame where it is safe to increase the severity. For Analyzers, this may mean tying it to the next value of AnalysisLevel (which is downstream of TFM). For small MSBuild and NuGet diagnostics, this may mean tying it to the next Warning Level or SdkAnalysisLevel. For CLI changes, this may mean tying it to the next LTS major version of the SDK. Ideally the way you would structure this increase would be automated and documented so that users know what's coming down the pipe.
+
+### Cut-over to new behavior after a long introduction period
+
+After the change has been introduced in a gradual way, cut over to the new behavior. This may mean removing the old behavior entirely, or it may mean making the new behavior the default and providing a way to opt out of it. It is important that you
+provide enough time for users to adapt - for example the `dotnet new --list` example above took an entire major release to make the new forms the default.
```

Comment:
Suggest a newline for each start of sentence to make future updates cleaner.

Reply:
I'm following the convention of other markdown in the repo.

Reply:
Git diffs on markdown work better with my suggestion. Up to you whether that's important.